src/wagtail_vector_index/storage/models.py

```diff
@@ -5,7 +5,7 @@
 from django.db.models import Q
 
 
-class DocumentQuerySet(models.QuerySet):
+class DocumentManager(models.Manager):
```

Comment:
Wouldn't the QuerySet derived manager be preferable here for allowing chaining custom operations + default queryset operations regardless of order? Not a big deal, but I'm not sure I get this change

Reply:
Yes definitely preferable but I ran in to typing issues using the `QuerySet` derived manager.

I think this is resolved in django-stubs by https://github.com/typeddjango/django-stubs/issues/738, but not in django-types which we're using in this project.